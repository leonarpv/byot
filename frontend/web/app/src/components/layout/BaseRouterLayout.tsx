import React from 'react';
import {Switch, Route, Redirect} from 'react-router-dom';
import {Router} from '../../router/Router';
import {LoginFormPage} from '../pages/LoginFormPage';
import {RegistrationFormPage} from '../pages/RegistrationFormPage';

interface Props {}

export const BaseRouterLayout: React.FC<Props> = (props: Props) => {
  return (
    <Switch>
      <Route exact path={Router.login.URI()}>
        <LoginFormPage />
      </Route>
      <Route exact path={Router.register.URI()}>
        <RegistrationFormPage />
      </Route>
      <Route exact path={Router.resetPassword.URI()}>
        <LoginFormPage />
      </Route>
      <Redirect to={Router.login.URI()} />
    </Switch>
  );
};
