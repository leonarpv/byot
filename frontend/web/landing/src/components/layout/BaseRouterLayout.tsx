import React from 'react';
import { Switch } from 'react-router-dom';

interface Props {}

export const BaseRouterLayout: React.FC<Props> = (props: Props) => {
  return <Switch></Switch>;
};
