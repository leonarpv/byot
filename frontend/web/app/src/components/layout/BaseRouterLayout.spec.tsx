import * as React from 'react';
import {BaseRouterLayout} from './BaseRouterLayout';
import {render, RenderResult} from '@testing-library/react';

describe('BaseRouterLayout', function () {
  let component: RenderResult;

  beforeEach(() => (component = render(<BaseRouterLayout />)));

  it('should render', () => expect(component.container).toMatchSnapshot());

  afterEach(() => jest.clearAllMocks());
});
