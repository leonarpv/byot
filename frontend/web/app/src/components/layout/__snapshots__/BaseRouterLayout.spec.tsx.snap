// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseRouterLayout should render 1`] = `
<div>
  <div
    data-testid="react-router-dom-switch"
  >
    <div
      data-testid="unauthorized-route-base"
    />
    <div
      data-testid="auth-route-base"
    />
    <div
      data-testid="react-router-dom-redirect"
    />
  </div>
</div>
`;
